### PR TITLE
chore: Split out google-cloud-errors gem

### PR DESCRIPTION
**Pull request by dazuma**
Proposal to split google-cloud-errors out of google-cloud-core, which will make it possible not to include core (with its legacy credentials class and config system) in microgenerator-based libraries.

### CODE REVIEW

File: google-cloud-errors/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+3. Install the top-level project dependencies.
+
+   ```sh
+   $ bundle install
+   ```
+
+4. Install the Google Cloud dependencies.
+
+   ```sh
+   $ cd google-cloud-core/
```
Comment (blowmage) [inline google-cloud-errors/CONTRIBUTING.md]: Change `google-cloud-core` to `google-cloud-errors` in this document.

File: google-cloud/Gemfile [hunk 2]
```diff
@@ -15,6 +15,7 @@ gem "google-cloud-dlp", path: "../google-cloud-dlp"
 gem "google-cloud-dns", path: "../google-cloud-dns"
 gem "google-cloud-env", path: "../google-cloud-env"
 gem "google-cloud-error_reporting", path: "../google-cloud-error_reporting"
+gem "google-cloud-errors", path: "../google-cloud-errors"
```
Comment (blowmage) [inline google-cloud/Gemfile]: Core should probably be listed here as well, to make sure it loads the correct version.

**Comment by blowmage** [comment]
Big :+1:!

File: google-cloud/Gemfile (hunk 2 above)
Comment (dazuma) [reply google-cloud/Gemfile]: Core is already listed here.

File: google-cloud-errors/CONTRIBUTING.md (hunk 1 above)
Comment (dazuma) [reply google-cloud-errors/CONTRIBUTING.md]: whoops, I knew I'd miss something. Done.

**Pull request merged**